aws/cloudformation/cloud_formation_stack.yml.erb

```diff
@@ -669,9 +674,13 @@ Resources:
           - !ImportValue VPC-ELBSecurityGroup # Accept HTTP traffic on daemon
 <%  end -%>
       DisableApiTermination: <%= rack_env?(:adhoc) ? false : true %>
-# We don't yet provision the production database cluster/instances via CloudFormation.
+    DependsOn:
+<%# We don't yet provision the production database cluster/instances via CloudFormation. %>
 <% if database && !rack_env?(:production) %>
-    DependsOn: DBProxy
+      - DBProxy
+<% end -%>
+<% if frontends-%>
+      - Memcached
 <% end -%>
```

Comment:
This could possibly generate an empty `DependsOn` array, which I believe is invalid YAML. You could use `!Ref AWS::NoValue` if you don't want to conditionally render the whole `DependsOn`

```yml.erb
<% if !frontends && !database -%>
      - !Ref AWS::NoValue
<% end -%>
```